docs/SILProgrammersManual.md

```diff
@@ -159,6 +159,401 @@ rules out PartialApplies is no excuse to conflate applied arguments
+the same as the access base for all storage kinds except global and
+class storage. For class properties, the storage root is the reference
+root of the object, not the base of the property. Multiple
+`ref_element_addr` projections may exist for the same property. Global
+variable storage is always uniquely identified, but it is impossible
+to find all uses from the def-use chain alone. Multiple `global_addr`
+instructions may reference the same variable. To find all global uses,
+the client must independently find all global variable references
+within the function. Clients that need to know which SILValue base was
+discovered during use-def traversal in all cases can make use of
+`AccessedStorageWithBase` or `AccessPathWithBase`. 
+
+### AccessPath
+
+`AccessPath` extends `AccessedStorage` to include the path components
+that determine the address of a subobject within the access base. The
+access path is a string of index offsets and subobject projection
+indices.
+
+```
+struct S {
+  var field0: Int64
+  var field1: Int64
+}
+
+%eltadr = struct_element_addr %access : $*S, #.field1
+
+Path: (#1)
+```
+
+```
+class A {}
+
+%tail  = ref_tail_addr %ref : $A, $S
+%one   = integer_literal $Builtin.Word, 1
+%elt   = index_addr %tail : $*S, %one : $Builtin.Word
+%field = struct_element_addr %elt : $*S, $S.field0
+
+Path: (@1, #0)
+```
+
+Note that a projection from a reference type to the object's property
+or tail storage is not part of the access path because it is already
+identified by the storage location.
+
+Offset indices are all folded into a single index at the head of the
+path (a missing offset implies offset zero). Offsets that are not
+static constants are still valid but are labeled "@Unknown". Indexing
+within a subobject is an ill-formed access, but is handled
+conservatively since this rule cannot be fully enforced.
+
+For example, the following is an invalid access path, which just
+happens to point to field1:
+```
+%field0 = struct_element_addr %base : $*S, #field0
+%field1 = index_addr %elt : $*Int64, %one : $Builtin.Word
+
+Path: (INVALID)
+```
+
+The following APIs determine whether an access path contains another
+or may overlap with another.
+
+`AccessPath::contains(AccessPath subPath)`
+
+`AccessPath::mayOverlap(AccessPath otherPath)`
+
+These are extremely light-weight APIs that, in the worst case, require
+a trivial linked list traversal with single pointer comparison for the
+length of subPath or otherPath.
+
+Subobjects are both contained with and overlap with their parent
+storage. An unknown offset does not contain any known offsets but
+overlaps with all offsets.
+
+### Access path uses
+
+For any accessed storage location and base, it must also be possible
+to reliably identify all uses of that storage location within the
+function for a particular access base. If the storage is uniquely
+identified, then that also implies that all uses of that storage
+within the function have been discovered.
```

Comment:
To clarify, you could add "In other words, there are no aliases to the use set within the function"

Reply:
Yes, I also added this comment
```
As passes begin to adopt AccessPath::collectUses(), I expect it to
become a visitor pattern that allows the pass to perform custom
book-keeping for certain types of uses.
```